scipy/interpolate/regrid_python.py

```diff
@@ -0,0 +1,1104 @@
+    fpold = None
+    last_axis = "y"
+    mpm = len(x) + len(y)
+    nx = len(tx)
+    ny = len(ty)
+    mx_head = max(0, nestx - nx) if nestx is not None else None
+    my_head = max(0, nesty - ny) if nesty is not None else None
+    fp0 = None
+    nplusx = None
+    nplusy = None
+
+    # https://github.com/scipy/scipy/blob/v1.16.2/scipy/interpolate/fitpack/fpregr.f#L51-L300
+    for it in range(mpm):
+        nx, ny = len(tx), len(ty)
+
+        (Ax, offset_x, nc_x,
+         Ay, offset_y, nc_y,
+         Drx, offset_dx, _,
+         Dry, offset_dy, _, Q) = build_matrices(
+             x_fit, y_fit, Z, tx, ty, kx, ky)
+        C0, fp  = _solve_2d_fitpack(Ax, offset_x, nc_x,
+                                    Drx, offset_dx,
+                                    Ay, offset_y, nc_y, Q,
+                                    Dry, offset_dy, -1,
+                                    kx, tx, x_fit, ky, ty,
+                                    y_fit, Z_fit)
+
+        if len(tx) == nminx and len(ty) == nminy:
+            fp0 = fp
+
+        if verbose:
+            print(f"[it={it}] p=0 fp={fp:.6e} s={s:.6e} "
+                  f"nx={nx} ny={ny} moves={moves}/{mpm}")
+            print(f"    headroom: mx_head={mx_head} my_head={my_head} "
+                  f"(nestx={nestx}, nesty={nesty})")
+
+        if s == 0.0:
+            if verbose:
+                print(f"  Target reached at it={it} (fp = {fp} <= s = {s})")
+            return return_NdBSpline(fp, (tx, ty, C0), (kx, ky))
+
+        if fp < s:
+            break
+
+        _Ax = BSpline.design_matrix(x_fit, tx, kx, extrapolate=False)
+        _Ay = BSpline.design_matrix(y_fit, ty, ky, extrapolate=False)
+        Z0  = (_Ax @ C0) @ _Ay.T
+        R = Z_fit - Z0
+
+        added_x = None
+        added_y = None
+        if last_axis == "y":
+            len_tx_before = len(tx)
+            tx, nplusx = _generate_knots(
+                x_fit, xb, xe, kx, s, nmin=nminx, nmax=nmaxx,
+                nest=nestx, t=tx, fp=fp, fpold=fpold,
+                residuals=np.ascontiguousarray(np.sum(R**2, 1)),
+                nplus=nplusx)
+            added_x = len(tx) - len_tx_before
+            if verbose:
+                print(f"    Inserted {added_x} knots in X")
+            last_axis = "x"
+        else:
+            len_ty_before = len(ty)
+            ty, nplusy = _generate_knots(
+                y_fit, yb, ye, ky, s, nmin=nminy, nmax=nmaxy,
+                nest=nesty, t=ty, fp=fp, fpold=fpold,
+                residuals=np.ascontiguousarray(np.sum(R**2, 0)),
+                nplus=nplusy)
+            added_y = len(ty) - len_ty_before
+            if verbose:
+                print(f"    Inserted {added_y} knots in Y")
+            last_axis = "y"
+
+        fpold = fp
+
+    if verbose:
+        print(f"Growth end with fp={fp:.6e} > s={s:.6e} "
+              f"({'nest exhausted' if (mx_head==0 and my_head==0) else 'early stop'})")
+
+    if len(tx) != nminx or len(ty) != nminy:
```

Comment:
What is this condition? IOW, under what conditions the algorithm have never added a single knot in any direction?